OpenSim/Simulation/Model/Frame.h

```diff
@@ -189,9 +189,13 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Frame, ModelComponent);
     ///@}
 
     /** Add a Mesh specified by file name to the list of Geometry owned by the Frame.
-    Scale defaults to 1.0 but can be changed on the call line. For convenience 
+    Scale defaults to 1.0 but can be changed on the call line for convenience.
     */
-    void addMeshGeometry(const std::string &aGeometryFileName, const SimTK::Vec3 scale = SimTK::Vec3(1));
+    void attachMeshGeometry(const std::string &aGeometryFileName, const SimTK::Vec3 scale = SimTK::Vec3(1));
+    /** Add a piece of Geometry to the list of Geometry owned by the Frame.
+    Scale defaults to 1.0 but can be changed on the call line, for convenience.
+    */
```

Comment:
I think the statement about scaling is evident from the function declaration. 

Could you add a bit to this doxygen documentation more clearly stating that this is a convenience over `ModelComponent::addGeometry()` for the case of adding geometry to a Frame, and that this sets the frame name? Also, a statement about memory management would be good too.
Perhaps something like:

> This function is a convenience for ModelComponent::addGeometry() for the case of adding Geometry directly to a Frame, and thus sets the "frame name" of the Geometry to *this* frame.  The provided `geom` is copied into this Frame, so any changes you make to `geom` after calling this method will not have an effect.